frontend/server/src/main/java/org/pytorch/serve/util/PriorityLinkedBlockingDeque.java

```diff
@@ -0,0 +1,151 @@
+package org.pytorch.serve.util;
+
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PriorityLinkedBlockingDeque<T extends Prioritisable> {
+
+    private static final Logger logger = LoggerFactory.getLogger(PriorityLinkedBlockingDeque.class);
+
+    // lock and condition for waiting on empty queues
+    final ReentrantLock lock = new ReentrantLock();
+    private final Condition notEmpty = lock.newCondition();
+
+    private int nPriorities;
+    private int sumPriorityWeights;
+    private int[] weightedPriorityMap;
+    private ConcurrentHashMap<Integer, LinkedBlockingDeque<T>> priorityDeques;
+
+    public PriorityLinkedBlockingDeque(int nPriorities, int queueSize) {
+
+        this.nPriorities = nPriorities;
+        this.priorityDeques = new ConcurrentHashMap<Integer, LinkedBlockingDeque<T>>();
+
+        // sum of 0 + 1 + 2 + ... + nPriorities - 1 via triangular sum
+        this.sumPriorityWeights = ((this.nPriorities - 1) * this.nPriorities) / 2;
```

Comment:
Note after our discussion: 

For our usecase with exactly two priorities, it might be worth to set the weights manually instead of using a formula.

Using a formula is probably cleaner if we want to add this upstream